miso-web/src/main/webapp/pages/editRun.jsp

```diff
@@ -256,9 +256,22 @@
         </thead>
         <tbody>
         <tr>
-          <td><fmt:formatDate pattern="dd/MM/yyyy" value="${run.startDate}"/></td>
           <c:choose>
-          <c:when test="${(run.health.isDone() and empty run.completionDate)}">
+          <c:when test="${run.id == 0 or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
+              <td><form:input path="startDate"/></td>
+          
+              <script type="text/javascript">
+              Utils.ui.addDatePicker("startDate");
+              </script>
+            </c:when>
+            <c:otherwise>
+              <td id="startDate">
+                <fmt:formatDate pattern="dd/MM/yyyy" value="${run.startDate}"/>
+              </td>
+            </c:otherwise>
+          </c:choose>
+          <c:choose>
+          <c:when test="${(run.health.isDone() and empty run.completionDate) or run.id == 0 or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
```

Comment:
Hmm, when I deployed to test, the completion date entry box was visible all the time so I don't know that there would have been a requirement to save twice. I think that before the big recent schema changes, there was an event handler to make the completion date box visible when the status is set to Failed or Completed. IMO, that would be the clearest way to implement it.